### PR TITLE
Add module metering

### LINKED ISSUE #19: Implement metering of modules
**Describe what you want implemented** 
The execution of modules should be metered. This can be done by using the facilities `wasmer` provides.

**Describe "Why" this is needed**
This is required to allow for charging for the execution of a smart contract.

**Describe alternatives you've considered**
Previously, metering was done using by annotating the code with the costs of each section. This was abandoned in favor of the more elegant approach used in `wasmer`.  

**Additional context**
This implementation should allow the user to call the virtual machine, specifying the amount of gas they're willing to spend. If the execution runs out, it should return an (OutOfGas) error, otherwise it should return the actual gas spent.

### PR DESCRIPTION
**Pull request by ureeves**
Resolves #19

### CODE REVIEW

File: hatchery/src/error.rs [hunk 1]
```diff
@@ -25,6 +27,7 @@ pub enum Error {
     MissingModuleExport,
     CompositeSerializerError(Compo),
     PersistenceError(io::Error),
+    OutOfPoints(ModuleId),
```
Comment (krl) [inline hatchery/src/error.rs:30]: I don't see the reason to include the ModuleId here

File: hatchery/src/world.rs [hunk 2]
```diff
@@ -31,13 +33,17 @@ use crate::snapshot::{MemoryPath, Snapshot, SnapshotLike};
...
+const POINT_PASS_PERCENTAGE: u64 = 93;
+
 #[derive(Debug)]
 pub struct WorldInner {
     environments: BTreeMap<ModuleId, Env>,
     storage_path: PathBuf,
     events: Vec<Event>,
     call_stack: CallStack,
     height: u64,
+    limit: u64,
```
Comment (krl) [inline hatchery/src/world.rs:46]: The point limit does not belong to the world.

**Review by krl — CHANGES_REQUESTED** [review]

File: hatchery/src/error.rs (hunk 1 above)
Comment (ureeves) [reply hatchery/src/error.rs:30]: The user might appreciate knowing which contract was given too little points to operate.

Comment (krl) [reply hatchery/src/error.rs:30]: This can already be more elegantly deducted from what's on top of the call stack

Comment (ureeves) [reply hatchery/src/error.rs:30]: The call stack is currently not exposed to the caller

Comment (krl) [reply hatchery/src/error.rs:30]: it feels off to me, but not a big deal.

File: hatchery/src/world.rs (hunk 2 above)
Comment (ureeves) [reply hatchery/src/world.rs:46]: Lets resolve this in favor of https://github.com/dusk-network/hatchery/issues/41

Comment (krl) [reply hatchery/src/world.rs:46]: postponed for later refactor.

**Review by krl — APPROVED** [review]

**Pull request merged**